src/main/resources/jenkins/plugins/office365connector/WebhookJobProperty/config.jelly

```diff
@@ -3,75 +3,98 @@
-                    </f:entry>
-                    <f:repeatableDeleteButton value="${%Delete}" />
-                </table>
-            </f:repeatable>
-        </f:entry>
-    </f:section>
+  				<table>
+					<f:entry field="webhook">
+
+						<f:entry title="Name" field="name">
+							<f:textbox name="name" value="${webhook.getName()}" />
+						</f:entry>
+
+						<f:entry title="URL" field="url">
+							<f:textbox name="url" value="${webhook.getUrl()}" />
+						</f:entry>
+
+						<f:advanced>
+                            <f:section title="Build status">
+                                <f:description>
+                                    This section defines for which build statuses the notification is sent.
+                                </f:description>
+
+                                <f:entry title="Notify Build Start">
+                                    <f:checkbox name="startNotification" value="true" checked="${webhook.isStartNotification()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Aborted">
+                                    <f:checkbox name="notifyAborted" value="true" checked="${webhook.isNotifyAborted()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Failure">
+                                    <f:checkbox name="notifyFailure" value="true" checked="${webhook.isNotifyFailure()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Not Built">
+                                    <f:checkbox name="notifyNotBuilt" value="true" checked="${webhook.isNotifyNotBuilt()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Success">
+                                    <f:checkbox name="notifySuccess" value="true" checked="${webhook.isNotifySuccess()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Unstable">
+                                    <f:checkbox name="notifyUnstable" value="true" checked="${webhook.isNotifyUnstable()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Back To Normal">
+                                    <f:checkbox name="notifyBackToNormal" value="true" checked="${webhook.isNotifyBackToNormal()}"/>
+                                </f:entry>
+
+                                <f:entry title="Notify Repeated Failure">
+                                    <f:checkbox name="notifyRepeatedFailure" value="true" checked="${webhook.isNotifyRepeatedFailure()}"/>
+                                </f:entry>
+                            </f:section>
+
+                            <f:section title="Advanced configuration">
+                                <f:entry title="Macros" field="macros">
+                                    <f:repeatable name="macros" var="macro"
```

Comment:
Yes, user can define as many macros as wishes and if at least one is matched notification is sent.